Automattic/wp-calypso
# Themes: Introduce getPurchaseUrl(), drop purchase action

Turns out we don't need that action since there's just a URL we can point to instead. Lets us get rid of another `action` that obscurely invoked `defer()` and `page()`.

To test:

* Make sure that purchasing a theme still works as before.
* Try both from the showcase and theme sheet.
* Try single-site, multi-site, and logged-out.

Loosely related to #8262, hence CC'ing @kwight.
